### ISSUE TITLE
custom price filter (no ranges)

### ISSUE THREAD

**Issue by undefinederror** [DivanteLtd/vue-storefront#4796]
## Prerequisites

Please answer the following questions for yourself before submitting an issue. 
- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

Hi, I am trying to implement a price filter slider.
So far I have done a component where custom ranges can be input, and that emits a `change` event with same payload as per `PriceSelector.vue`.
It seems that the filter is applied only when `from` and `to` happen to form an existing `priceFilter range`

So for instance, this works
![image](https://user-images.githubusercontent.com/8209966/91552629-69bc4380-e92c-11ea-99ac-5cb69cf499c5.png)

While this does nothing at all.
![image](https://user-images.githubusercontent.com/8209966/91552753-9d976900-e92c-11ea-8756-f3b59bd0c1e6.png)

Why is that and how can I go about it?

Thank you

**Comment by undefinederror** [comment]
I managed to get it to work, luckily without overriding anything in core. 

For reference to others who may need this, I only edited Category.vue

![image](https://user-images.githubusercontent.com/8209966/91592620-17484a80-e95f-11ea-841f-52f0bc0e5f39.png)

Basically, since filters have to match what is in `store.getters['category-next/getFiltersMap'][category.id]` I go and override the price range array every time I change the price filter.

If anyone has a better idea of how to do it better please let me know.

**Comment by ThermScissorpunch** [comment]
I also tried this, but when you reload the page the filter is gone. 
Did you get it working there also?

**Comment by undefinederror** [comment]
yes. the price filter is added in query string and survives page refresh.
I don't think I ever changed much about that ever since I did it, and we are now in prod with this.

**Comment by Sproge-Kostya** [comment]
I would like to know the status on this issue. I have the same task now, I need to make a price range slider, if you found a solution. I would be grateful and I think in general this is a useful functionality

**Comment by dimasch** [comment]
Maybe someone will help, tested on VSF core `v1.12.2` with Capybara `v1.0.2`:

```
import config from 'config';
...

const adjustFiltersMapsForPrice = (category, filters, store) => {
  const categoryFilters = store.getters['category-next/getFiltersMap'][category.id];
  const priceObj = filters.price.split('-');
  const minPrice = Math.round(priceObj[0]);
  const maxPrice = Math.round(priceObj[1]);

  const newOne = { id: filters.price, type: 'price', from: minPrice, to: maxPrice, label: '', single: true };
  const newPriceFilters = [...categoryFilters.price, newOne];

  const newCategoryFilters = {...categoryFilters, price: newPriceFilters }
  store.commit('category-next/category/SET_CATEGORY_FILTERS', { category, filters: newCategoryFilters });
}


const composeInitialPageState = async (store, route, forceLoad = false) => {
...
  const currentFilters = route[config.products.routerFiltersSource];
  if (currentFilters.price) adjustFiltersMapsForPrice(currentCategory, currentFilters, store);

  const breadCrumbsLoader = store.dispatch(....
}
```
